init/init.ts

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env node
+import fs from 'fs'
+import path from 'path'
+import chalk from 'chalk'
+import findUp from 'find-up'
+import inqueier from 'inquirer'
+import highlight from 'cli-highlight'
+import { Template } from './Template'
+import { NextTemplate } from './templates/next'
+import { WebpackTemplate } from './templates/webpack-file'
+import { ReactScriptsTemplate } from './templates/react-scripts'
+import { BabelTemplate } from './templates/babel'
+import { WebpackOptions } from './templates/webpack-options'
+
+const templates: Record<string, Template<any>> = {
+  'next.js': NextTemplate,
+  'create-react-app': ReactScriptsTemplate,
+  webpack: WebpackTemplate,
+  babel: BabelTemplate,
+  'default (webpack options)': WebpackOptions,
+}
+
+type TemplateGuess<T> = {
+  defaultTemplate: Template<T> | null
+  defaultTemplateName: string | null
+  templatePayload: T | null
+}
+
+export function guessTemplateForUsedFramework<T>(): TemplateGuess<T> {
+  for (const [name, template] of Object.entries(templates)) {
+    const typedTemplate = template as Template<T>
+    const { success, payload } = typedTemplate.test(process.cwd())
+
+    if (success) {
+      return {
+        defaultTemplate: typedTemplate,
+        defaultTemplateName: name,
+        templatePayload: payload ?? null,
+      }
+    }
+  }
+
+  return {
+    templatePayload: null,
+    defaultTemplate: null,
+    defaultTemplateName: null,
+  }
+}
+
+async function getCypressConfig() {
+  const cypressJsonPath = await findUp('cypress.json')
+
+  // TODO figure out how to work with newly installed cypress
+  if (!cypressJsonPath) {
+    console.log(
+      `\nIt looks like you did not install cypress, we were unable to find ${chalk.green(
+        'cypress.json',
+      )} in this or parent directories. Please install cypress, run ${chalk.inverse(
+        ' cypress open ',
+      )} and rerun this script.`,
+    )
+
+    console.log(
+      `\nFind more information about installation and init script: ${chalk.bold.underline(
+        'https://github.com/bahmutov/cypress-react-unit-test#init',
+      )}`,
+    )
```

Comment:
```suggestion
    console.log(
      `\nIt looks like Cypress is not installed because we were unable to find ${chalk.green(
        'cypress.json',
      )} in this project. Please install Cypress via ${chalk.inverse( yarn add cypress -D )}` (or npm), and then run ${chalk.inverse(
        ' cypress open ',
      )} and rerun this script.`,
    )

    console.log(
      `\nFind more information about installation at: ${chalk.bold.underline(
        'https://github.com/bahmutov/cypress-react-unit-test#init',
      )}`,
    )
```